### PR TITLE
[src/platform] Separation of nrfconnect and Zephyr platforms

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
#### Problem
nrfconnect platform is based on the Zephyr and in many cases it uses Zephyr specific components implementation. Zephyr specific files should be located in the Zephyr platform.

 #### Summary of Changes
* Moved Zephyr specific modules from the nrfconnect to the Zephyr platform
* Moved BLEManagerImpl and ThreadStackManagerImpl implementations from Generic_Zephyr files.
* Removed Generic_Zephyr files.
* Aligned READMEs for nrfconnect and Zephyr platforms.

### CODE REVIEW

**Comment by turon** [comment]
Is there any reason that splitting out nrfconnect-specific code needs to also remove the generic Zephyr files?

**Comment by Damian-Nordic** [comment]
@turon we realized that some of nrfconnect platform classes only refer to Zephyr API, so they can potentially be reused by other Zephyr-based platforms. Generic..._Zephyr classes are removed since we can't think of a real use case for them which would justify more complicated class hierarchy they bring (given that Zephyr provides a nice abstraction layer over hardware drivers which significantly reduces chances that platform-specific additions will be needed for the CHIP platform layer components like BLE, Thread or Configuration manager). Do you think we should rather extract the code to Generic...Zephyr classes?

**Review by bzbarsky-apple — APPROVED** [review]

File: src/platform/Zephyr/CHIPDevicePlatformEvent.h [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+#pragma once
+
+#include <platform/CHIPDeviceEvent.h>
+
+#include <bluetooth/bluetooth.h>
+
+namespace chip {
+namespace System {
+class PacketBuffer;
+}
```
Comment (mspang) [inline src/platform/Zephyr/CHIPDevicePlatformEvent.h:33]: // namespace System

**Review by mspang — APPROVED** [review]

**Comment by kkasperczyk-no** [comment]
@jelderton @saurabhst @BroderickCarlin @chrisdecenzo could somebody of you take a look on this PR and review it?

**Review by LuDuda — APPROVED** [review]
LGTHM :+1: Thanks!

**Review by turon — APPROVED** [review]
LGTM 👍

**Review by rwalker-apple — APPROVED** [review]

**Review by saurabhst — APPROVED** [review]

**Pull request merged**